openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx

```diff
@@ -165,57 +162,47 @@ export const AuthProvider = ({
         }
       })
       .catch((err: AxiosError) => {
-        if (err.response?.data.code === 404) {
-          resetUserDetails();
+        resetUserDetails();
+        if (err.response?.data.code !== 404) {
+          showErrorToast(
+            err,
+            jsonData['api-error-messages']['fetch-logged-in-user-error']
+          );
         }
       });
   };
 
-  const getUpdatedUser = (data: User, user: OidcUser) => {
-    let getAdminCookie = localStorage.getItem(isAdminUpdated);
-
-    // TODO: Remove when using cookie no more
-    if (!getAdminCookie) {
-      getAdminCookie = cookieStorage.getItem(isAdminUpdated);
-      if (getAdminCookie) {
-        localStorage.setItem(isAdminUpdated, getAdminCookie);
-      }
-    }
-
-    if (getAdminCookie) {
-      appState.updateUserDetails(data);
-    } else {
-      const updatedData = {
-        isAdmin: data.isAdmin,
-        email: data.email,
-        name: data.name,
-        displayName: user.profile.name,
-        profile: { images: getImages(user.profile.picture ?? '') },
-      };
-      updateUser(updatedData)
-        .then((res: AxiosResponse) => {
+  const getUpdatedUser = (updatedData: User, existingData: User) => {
+    const { isAdmin, name, displayName, profile, email } = updatedData;
+    updateUser({ isAdmin, name, displayName, profile, email })
+      .then((res: AxiosResponse) => {
+        if (res.data) {
           appState.updateUserDetails(res.data);
-          localStorage.setItem(isAdminUpdated, 'true');
-        })
-        .catch((error: AxiosError) => {
-          showErrorToast(
-            error,
-            jsonData['api-error-messages']['update-admin-profile-error']
-          );
-        });
-    }
+        } else {
+          throw jsonData['api-error-messages']['unexpected-server-response'];
+        }
+      })
+      .catch((error: AxiosError) => {
+        appState.updateUserDetails(existingData);
+        showErrorToast(
+          error,
+          jsonData['api-error-messages']['update-admin-profile-error']
+        );
+      });
   };
 
   const handleSuccessfulLogin = (user: OidcUser) => {
     setLoading(true);
     getUserByName(getNameFromEmail(user.profile.email), userAPIQueryFields)
       .then((res: AxiosResponse) => {
         if (res.data) {
-          if (res.data?.isAdmin) {
-            getUpdatedUser(res.data, user);
+          const updatedUserData = getUserDataFromOidc(res.data, user);
+          if (!matchUserDetails(res.data, updatedUserData, ['profile'])) {
```

Comment:
If previously we had a profile picture, but the update from SSO gives a blank string for profile picture, we should retain the old one and not update with empty string.